docs/core/deploying/single-file.md

```diff
@@ -0,0 +1,96 @@
+
+Single-file will not bundle native libraries by default. On Linux, we prelink the runtime into the bundle and only application native libraries are left next to the bundle. On Windows, we prelink only the hosting code and both the runtime and application native libraries are left next to the bundle. This is to ensure a good debugging experience, which requires native files to be excluded from the single file. There is an option to set a flag, `IncludeNativeLibrariesForSelfExtract`, to include native libraries in the single file bundle, but these files will be extracted to a temporary directory in the client machine when the single file application is run.
+
+## Exclude files being embedded in Single File
+
+Certain files can be explicitly excluded from being embedded in the single-file by setting following meta-data:
+
+```xml
+<ExcludeFromSingleFile>true</ExcludeFromSingleFile>
+```
+
+For example, to place some files in the publish directory but not bundle them in the single-file:
+
+```xml
+<ItemGroup>
+    <Content Update="Plugin.dll">
+      <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
+      <ExcludeFromSingleFile>true</ExcludeFromSingleFile>
+    </Content>
+  </ItemGroup>
+
+```
+
+## Publish a Single File app - CLI
+
+Publish a single file application using the [dotnet publish](../tools/dotnet-publish.md) command. When you publish your app, set the following properties:
+
+- Publish as a self-contained app for a specific runtime: `-r win-x64`
+- Publish as a single-file: `-p:PublishSingleFile=true`
+
+The following example publishes an app for Windows as a self-contained single file application.
+
+```dotnet
+dotnet publish -r win-x64 -p:PublishSingleFile=true --self-contained true
+```
+
+The following example publishes an app for Linux as a framework dependent single file application..
+
+```dotnet
+dotnet publish -r linux-x64 -p:PublishSingleFile=true --self-contained false
+```
+
+For more information, see [Publish .NET Core apps with .NET Core CLI](deploy-with-cli.md).
+
+## Publish a Single File app - Visual Studio
+
+Visual Studio creates reusable publishing profiles that control how your application is published.
+
+01. On the **Solution Explorer** pane, right-click on the project you want to publish. Select **Publish...**.
+
+    :::image type="content" source="media/trim-self-contained/visual-studio-solution-explorer.png" alt-text="Solution Explorer with a right-click menu highlighting the Publish option.":::
+
+    If you don't already have a publishing profile, follow the instructions to create one and choose the **Folder** target-type.
+
+01. Choose **Edit**.
+
+    :::image type="content" source="media/trim-self-contained/visual-studio-publish-edit-settings.png" alt-text="Visual studio publish profile with edit button.":::
```

Comment:
- ⚠️ Don't use images from other articles, copy the image to the appropriate folder for this article.

Reply:
@adegeo FYI, I believe I recently met some code snippets or images referenced from other articles in the repo.